Winston slack

JWSong opened the pull request:
With this PR, `winston` will send errors to Slack `#general` channel as well.

I tried to use [winston-slack](https://github.com/TDASinternal/winston-slack), but it is broken and not maintained for more than one year.
Thus, I got it fixed and included the fixed one.

We need following requirements to make this PR work:
- One more env variable, `SLACK_WEBHOOK_URI`, must be set.
- Our Slack team must set up [incoming webhook integration](https://api.slack.com/incoming-webhooks).

We can discuss about a proper channel for these logs, text-format, bot's name, and so on.
Since I do not have much experience with js yet, please review my code and feel free to point out my mistakes and bad practices.

Commit:
Added a transport to slack in winston.
Now, error logs are sent to Slack #general channel as well.
I tried to use winston-slack module, but they are broken now and not maintaining the code.
Thus, I got it fixed and included here.

Commit:
Added few more rules.

`.idea` is for JetBrain IDE users.
`*.log.*` is for npm logs.

Commit:
Added information about Slack webhook in readme

JWSong committed:
Merge branch 'development' of github.com:dvcoders/dvcoders-backend into winston-slack

thebearjew commented:
Some of the details for up for discussion:

- Log location: #server-logs (we've discussed this one)
- Bot's name: dvserver
- Image: dvcoders logo

JWSong committed:
changed bot's name and channel to send a msg

JWSong committed:
Changed the errorbot's icon to the dvcoder icon
I got the dvcoder icon from the github avartar url.
If you have better idea to get the icon, please make a comment to discuss.

JWSong commented:
Completed your request.
 - Log location: #server-logs
 - Bot's name: dvserver
 - Image: dvcoders logo

I got the dvcoders logo from dvcoders github avatar. If it is not appropriate, please let me know and discuss about it.

JWSong committed:
Merge branch 'development' of github.com:dvcoders/dvcoders-backend into winston-slack

JWSong committed:
Observe the js standard

janka102 reviewed (changes requested):
It looks like the merge from the current dev branch overwrote the changes of adding `SLACK_WEBHOOK_URI` into the README and removed some dependencies in `package.json`. If you could add those back that'd be great, otherwise good so far! 👍

janka102 commented:
Also, just created a webhookuri a default username, icon, and channel can be configured in Slack itself, but the request can override it like you do. I was thinking maybe it’s better to just use the defaults from Slack and not need to send that in the requests. What you and others think?

JWSong committed:
Added the webhook uri env var information to README

JWSong committed:
Added a dependancy (node-slack)

JWSong committed:
Typo in README
